NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers
# Ticket #7 - Delete a Post

# Description

**Completed ticket #7 [Delete a Post](https://trello.com/c/4pZWHZ7W/7-7-delete-a-post)
Added functionality to the Delete button on Posts. 
## Type of change

# Testing Instructions

**GIT this branch:**

1. Clone this repository to your own machine using `[git clone https://github.com/NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers.git](https://github.com/NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers/tree/main/SQL) `
2. `git fetch -a` in the master branch to retrieve this branch named bh-deletePost.
3.  checkout this branch using the command `git checkout bh-deletePost`

**Create the database:**

1. Open Visual Studio.
2. At the start screen select "Open a Local Folder"
4. Navigate to your workspace directory, open the `tabloidmvc-yellow-belly-yackers` folder, then select the SQL folder
5. Run both of these SQL files [01_TabloidMVC_Create_DB.sql 02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers/tree/main/SQL)

**Now you need to add data to test the delete function for posts.** 

> 1. In Visual Studio, Open the View menu and select SQL Server Object Explorer.
> 2. Expand the node for your SQL Server instance.
> 3. Expand "Databases"  
> 4. Right-click on the TabloidMVC database and choose New Query.
> 5. Run this SQL script to add a post to test the delete function.

> INSERT INTO [Post] ([Id], [Title], [Content], [ImageLocation], [CreateDateTime], [PublishDateTime], [IsApproved], [CategoryId], [UserProfileId])
VALUES (1, 'TEST DATA TO DELETE', 'GOING TO DELETE THIS.', 'https://gizmodiva.com/wp-content/uploads/2017/10/SCOTT-A-WOODWARD_1SW1943-1170x689.jpg',SYSDATETIME(), SYSDATETIME(), 1, 1, 1);

5. Now you have a post to delete! 

**Test the new feature:**

1.	Click the Run button in visual studio to launch the site. 
2.	In the header, next to the Title (Tabloid) there should be a single link to Login, click this.
3.	For testing purposes, log in as admin@example.com, and click login.
4.	After successfully logging in, you should now see a link to MYPOSTS, click this.
5.	You will see the new data you added titled TEST DATA TO DELETE at the top of the list above C# is the Best Language.
6.	Click the trashcan button to delete.
7.	This will take you to the Delete confirmation page. Where you will have a choice to delete or go back to list. Click delete to test if it works. 
8.	That will redirect you back to the list of posts and the Post titled TEST DATA TO DELETE should be gone.



# Checklist:
